enhancements/sig-architecture/132-work-update-only-when-spec-change/README.md

```diff
@@ -0,0 +1,183 @@
+- [] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+This proposal is to introduce new options in update stratgey of manifestwork to resolve issues
+on how to coordinate other actors and work agent when they apply the same resource on the spoke
+cluster.
+
+## Motivation
+We have introduced several update strategy in `ManifestWork` to resolve resource conflict before:
+- `CreateOnly` let work-agent to create the resource only and ignore any further change on the
+resource
+- `ServerSideApply` utilize the server side apply mechanism in kube-apiserver, so if work-agent
+and another actor try to operator the same field in a resource, a conflict can be identified.
+
+However, some issues are raised indicating the above strategy can still not meet all the requirements
+- [issue 631](https://github.com/open-cluster-management-io/ocm/issues/631) reports a case that user
+uses manifestwork to create a configmap, and then another actor on the spoke change the configmap. The
+user does not want the configmap to be changed back by the work-agent. However, when the configmap resource
+in the manifestwork is updated, the user wants the configmap to be updated accordingly.
+- [issue 670](https://github.com/open-cluster-management-io/ocm/issues/670) is a case when the user
+uses manifestwork to create argocd's `Application` resource. The `Application` resource has an `operation`
+field which is used to trigger the sync, and the field will be removed by the argocd when the sync is done.
+User in the manifestwork sets the `operation` field and when argocd removes the field, the user does not want
+the field to be updated back by the work-agent.
+- [issue 690](https://github.com/open-cluster-management-io/ocm/issues/690) is the case that user wants
+to create a deployment using manifestwork, but what HPA on the spoke the control the replicas of the deployment.
+Hence the user wants the work-agent to ignore the replicas field of the deployment in the manfiestwor if it is
+set.
+
+
+## Proposal
+
+We would like to introduce new options in `Update` and `ServerSideApply` update strategy to resolve the above issue.
+So user can set the option in the manifestwork, so when another actor on the spoke cluster update the resource. The
+work-agent will ignore the change and not try to change the resource back. On the other hand, when the resource spec
+defined in the manifestwork is updated by the user, the work-agent will still update the resource accordingly. In
+summary, the option is to ignore the change triggered from spoke cluster only.
+
+In addition, in `ServerSideApplly` strategy, we would also introduce a new `ignoreDifferences` options similar as what
+is defined in argoCD (https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/). Such that user can choose to not
+let work-agent patch certain resource fields.
```

Comment:
This will also be able to resolve the zero creationTime in metadata problem, right?